src/main/java/io/debezium/connector/jdbc/PreparedStatementQueryBinder.java

```diff
@@ -21,7 +22,12 @@ public void bind(ValueBindDescriptor valueBindDescriptor) {
 
         try {
             if (valueBindDescriptor.getTargetSqlType() != null) {
-                binder.setObject(valueBindDescriptor.getIndex(), valueBindDescriptor.getValue(), valueBindDescriptor.getTargetSqlType());
+                if (valueBindDescriptor.getTargetSqlType() == Types.ARRAY) {
+                    binder.setArray(valueBindDescriptor.getIndex(), valueBindDescriptor.getArray(binder.getConnection()));
```

Comment:
So I would recommend that we do away with the `ValueBindDescriptorArray` entirely and just inline the `getArray` logic here. We're already predicting this block on the `Types.ARRAY` trait to call `binder.setArray`, so I think the extra class is really unnecessary.  We can simply add one additional argument to its constructor called `elementType` that would only be used in collection-based value bindings.  

This aligns with what we already when using the index/value versus index/value/target-type.

This also means we leave the access to the connection from the binder in these classes.

Reply:
I need the subtype (`text` or `bool` for example). How to do I extract here? New field in `ValueBindDescriptor` ?

Reply:
Pushed a new version @Naros